bundles/AdminBundle/Controller/Admin/DataObject/DataObjectController.php

```diff
@@ -567,12 +569,46 @@ static function (Task $task) {
 
             DataObject\Service::removeElementFromSession('object', $object->getId());
 
+
+            $layoutArray = json_decode($this->encodeJson($data['layout']), true);
+            $this->classFieldDefinitions = json_decode($this->encodeJson($object->getClass()->getFieldDefinitions()), true);
+            $this->injectValuesForCustomLayout($layoutArray);
+            $data['layout'] = $layoutArray;
+
             return $this->adminJson($data);
         }
 
         throw $this->createAccessDeniedHttpException();
     }
 
+    private function injectValuesForCustomLayout(array &$layout): void
+    {
+        foreach ($layout['children'] as &$child) {
+            if ($child['datatype'] === 'layout') {
+                $this->injectValuesForCustomLayout($child);
+            } else {
+                foreach ($this->classFieldDefinitions[$child['name']] as $key => $value) {
+                    if ($child[$key] === null || $child[$key] === '' || (is_array($child[$key]) && empty($child[$key]))) {
+                        $child[$key] = $value;
+                    }
+                }
+            }
+        }
+
+        //TODO remove in Pimcore 11
+        foreach ($layout['childs'] as &$child) {
+            if ($child['datatype'] === 'layout') {
+                $this->injectValuesForCustomLayout($child);
+            } else {
+                foreach ($this->classFieldDefinitions[$child['name']] as $key => $value) {
+                    if ($child[$key] === null || $child[$key] === '' || (is_array($child[$key]) && empty($child[$key]))) {
+                        $child[$key] = $value;
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Maybe you could combine this into one function and pass either children (for 11)  or childs